src/adapters/xhb.js

```diff
@@ -0,0 +1,187 @@
+                    sizeQueryString = sizeQueryString.slice(0, sizeQueryString.length - 1);
+                }
+            }
+        }
+
+        if (sizeQueryString) {
+            jptCall += sizeQueryString + '&';
+        }
+
+
+
+
+        //append custom attributes:
+        var paramsCopy = utils.extend({}, bid.params);
+
+        //delete attributes already used
+        delete paramsCopy.placementId;
+        delete paramsCopy.invCode;
+        delete paramsCopy.query;
+        delete paramsCopy.referrer;
+        delete paramsCopy.alt_referrer;
+
+        //get the reminder
+        var queryParams = utils.parseQueryStringParameters(paramsCopy);
+
+        //append
+        if (queryParams) {
+            jptCall += queryParams;
+        }
+
+        //append referrer
+        if (referrer === '') {
+            referrer = utils.getTopWindowUrl();
+        }
+
+        jptCall = utils.tryAppendQueryString(jptCall, 'referrer', referrer);
+        jptCall = utils.tryAppendQueryString(jptCall, 'alt_referrer', altReferrer);
+
+
+
+
+        //remove the trailing "&"
+        if (jptCall.lastIndexOf('&') === jptCall.length - 1) {
+            jptCall = jptCall.substring(0, jptCall.length - 1);
+        }
+
+        // @if NODE_ENV='debug'
+        utils.logMessage('jpt request built: ' + jptCall);
+        // @endif
+
+        //append a timer here to track latency
+        bid.startTime = new Date().getTime();
+
+        return jptCall;
+    }
+
+    //expose the callback to the global object:
+    $$PREBID_GLOBAL$$.handleXhbCB = function (jptResponseObj) {
+
+        var bidCode;
+
+        if (jptResponseObj && jptResponseObj.callback_uid) {
+
+            var responseCPM;
+            var id = jptResponseObj.callback_uid;
+            var placementCode = '';
+            var bidObj = getBidRequest(id);
+            if (bidObj) {
+                bidCode = bidObj.bidder;
+                placementCode = bidObj.placementCode;
+                //set the status
+                bidObj.status = CONSTANTS.STATUS.GOOD;
+            }
+
+            // @if NODE_ENV='debug'
+            utils.logMessage('JSONP callback function called for ad ID: ' + id);
+            // @endif
+
+            var bid = [];
+            if (jptResponseObj.result && jptResponseObj.result.ad && jptResponseObj.result.ad !== '') {
+
+
+                //responseCPM = parseInt(jptResponseObj.result.cpm, 10);
+                //CPM response from /jpt is dollar/cent multiplied by 10000
+                //in order to avoid using floats
+                //switch CPM to "dollar/cent"
+                //responseCPM = responseCPM / 10000;
+                responseCPM = 0.00;
+
+                //store bid response
+                //bid status is good (indicating 1)
+
+                var adId = jptResponseObj.result.creative_id;
+                bid = bidfactory.createBid(1);
```

Comment:
Please pass through the original bid request object (`bidObj`) as the 2nd param to `createBid`.
`bid = bidfactory.createBid(CONSTANTS.STATUS.GOOD, bidObj);`